app/weather_forecast.py

```diff
@@ -0,0 +1,265 @@
+
+
+def get_forecast_weather(input_date, location):
+    """ get the relevant weather forecast by calling the API.
+    Args:
+        input_date (date) - day part of date.
+        location (str) - location name.
+    Returns:
+        (int) - minimum degrees in Celsius.
+        (int) - maximum degrees in Celsius.
+        (str) - weather conditions.
+        (str) - location / error description.
+    """
+    input_query_string = BASE_QUERY_STRING
+    input_query_string["location"] = location
+    success_in_get_weather_data, api_json = get_data_from_api(FORECAST_URL, input_query_string)
+    if success_in_get_weather_data:
+        for item in api_json:
+            for i in range(len(api_json[item]['values'])):
+                if input_date == datetime.fromisoformat(api_json[item]['values'][i]['datetimeStr'][:-6]):
+                    min_temp = api_json[item]['values'][i]['mint']
+                    max_temp = api_json[item]['values'][i]['maxt']
+                    conditions = api_json[item]['values'][i]['conditions']
+        return min_temp, max_temp, conditions, api_json[item]['address']
+    else:
+        return None, None, None, api_json['message']
+
+
+def get_relevant_years_for_historical_average(day, month):
+    """ get a list for relevant years to call the get_historical_weather function
+        according to if date occurred this year or not.
+    Args:
+        day (int) - day part of date.
+        month (int) - month part of date.
+    Returns:
+        (list) - relevant years range.
+    """
+    if datetime.now() > datetime(year=datetime.now().year, month=month, day=day):
+        last_year = datetime.now().year
+    else:
+        last_year = datetime.now().year - 1
+    return list(range(last_year, last_year - HISTORICAL_AVERAGE_NUM_OF_YEARS, -1))
+
+
+def get_historical_average_weather(day, month, location):
+    """ get historical average weather by calling the get_historical_weather function
+        several times and calculate average.
+    Args:
+        day (int) - day part of date.
+        month (int) - month part of date.
+        location (str) - location name.
+    Returns:
+        (int) - minimum average degrees in Celsius.
+        (int) - maximum average degrees in Celsius.
+        (str) - location / error description.
+    """
+    sum_min = 0
+    sum_max = 0
+    if day == 29 and month == 2:
+        day = 28
+    relevant_years = (get_relevant_years_for_historical_average(day, month))
+    for relevant_year in relevant_years:
+        input_date = datetime(year=relevant_year, month=month, day=day)
+        min_temp, max_temp, conditions, description = get_historical_weather(input_date, location)
+        if min_temp is not None:
+            sum_min += min_temp
+            sum_max += max_temp
+    if min_temp is not None:
+        return sum_min / HISTORICAL_AVERAGE_NUM_OF_YEARS, sum_max / HISTORICAL_AVERAGE_NUM_OF_YEARS, description
+    else:
+        return None, None, description
```

Comment:
Remove the extra parenthesis and place is directly in the for.
`for relevant_year in get_relevant_years_for_historical_average(day, month):`

If get_historical_weather(input_date, location) returns an error, such as a site down, bad api key, invalid address, then the function will still call the API x amount of times. Should it?